### DISCUSSION TITLE
Moving UUID & ULID utilities to its own class

### DISCUSSION THREAD

**Discussion by DarkGhostHunter** [laravel/framework#47044] [General]
Title. Most of UUID utilities are being shoved into Str.

I think that, to avoid the `Str` helper creeping with more UUID related utilities, a new `UID` utility class should exist to wrap these helpers and future helpers.

Since this kind of update would be disruptive, `Str` would proxy these calls to `Uid`.

**Comment by shaedrich** [comment]
Same with the "case" methods:

* `Case::camel()`
* `Case::headline()`
* `Case::kebab()`
* `Case::lcfirst()`
* `Case::lower()`
* `Case::snake()`
* `Case::title()`
* `Case::ucfirst()`
* `Case::upper()`

One thing that bugs me with the bloated [`Str` class](https://laravel.com/docs/10.x/strings), is that generator (`uuid()`, `random()`), validator (`isUuid()`, `startsWith()`), manipulator (`mask()`) and render (`markdown()`) functions are mixed.

**Reply by DarkGhostHunter** [reply]
Yeah, seems that the-bloating Str with cases could also good.

**Reply by shaedrich** [reply]
I re-ordered and re-grouped the documentation page and added sub classes to illustrate what I mean (WIP):
https://gist.github.com/shaedrich/fb6c29016361db03736fa4541858e3d3

**Reply by DarkGhostHunter** [reply]
That's just plain better. What @taylorotwell thinks about this?

**Reply by shaedrich** [reply]
I'll give Taylor the favor of a doubt, however, there's a reason, I'm not submitting any PRs.

**Reply by shaedrich** [reply]
#52979 is why
